### PR TITLE
Allow DefaultCloudSpec to work by taking a pointer

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Currently, `DefaultCloudSpec` never does anything, because it takes a literal `kubermaticv1.CloudSpec` and only returns an error, meaning all changes to the `CloudSpec` will never leave the `DefaultCloudSpec` func.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

Cant unsee it now that you mentioned it :see_no_evil: 

/assign @nikhita 
CC @mrIncompetent

### CODE REVIEW

File: api/pkg/provider/cloud/aws/provider.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ type amazonEc2 struct {
 	dcs map[string]provider.DatacenterMeta
 }
 
-func (a *amazonEc2) DefaultCloudSpec(spec kubermaticv1.CloudSpec) error {
+func (a *amazonEc2) DefaultCloudSpec(spec *kubermaticv1.CloudSpec) error {
```
Comment (nikhita) [inline api/pkg/provider/cloud/aws/provider.go:40]: Should we also move the defaulting logic from `InitializeCloudProvider` to this (or maybe in a followup)?

**Review by nikhita — COMMENTED** [review]
> Cant unsee it now that you mentioned it

:smile:

File: api/pkg/provider/cloud/aws/provider.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/provider/cloud/aws/provider.go:40]: I'd prefer a follow-up

**Review by nikhita — APPROVED** [review]
/lgtm
/approve

**Pull request merged**